### DISCUSSION TITLE
How to use MAX31865 and PT100?

### DISCUSSION THREAD

**Discussion by werner53844** [arendst/Tasmota#15046] [Q&A] [answered]
Helle,
I want to use a PT100 (4-Wire) with a MAX31865 connected to a WEMOS D1 mini.
CS -> D3 / GPIO0
SDI -> D6 / GPIO12
SDO -> D7 / GPIO13
CLK -> D5 / GPIO14
GND -> GND
Vin -> 3V3
![Screenshot 2022-03-06 211839](https://user-images.githubusercontent.com/47952051/156940639-e5879676-9dcc-4acd-b42b-ce145c308946.jpg)

If I load this Arduino-Sketch:

[max31865_Test_WiFi_MQTT.txt](https://github.com/arendst/Tasmota/files/8193414/max31865_Test_WiFi_MQTT.txt)

it works correct and shows the right temperatur.

If I upload a self-compiled Tasmota-bin-File with following parameters:
#define FRIENDLY_NAME       "ndkpc195"
#define MQTT_USE            true
#define MQTT_HOST           "10.178.2.3"
#define MQTT_PORT           1893
#define MQTT_USER           "werner"
#define MQTT_PASS           "terminal"
#define NTP_SERVER1         "10.178.2.1"
#define USE_DS18x20
#define USE_SPI
#define USE_MAX31865
#define MAX31865_PTD_WIRES  4
#define MAX31865_PTD_RES    100
#define MAX31865_REF_RES    430
#define MAX31865_PTD_BIAS   0
#define TELE_PERIOD         60
it show a temperatur like:
tele/tasmota_34D008/SENSOR = {"Time":"2022-03-06T20:40:57","MAX31865-0":{"Temperature":988.8,"Resistance":430.0,"Error":0},"TempUnit":"C"}

Where do I think wrong?

Veryy much for your support.
Werner

**Comment by curzon01** [comment]
If you wired D3 / GPIO0 as CS then define D3 / GPIO0 as MAX31865 as CS1, not D4 / GPIO2

BTW, doesn't matter but it's unnecessary to define because these are the defaults:
```C
#define MAX31865_PTD_RES 100
#define MAX31865_REF_RES 430
#define MAX31865_PTD_BIAS 0
```

**Comment by werner53844** [comment]
Hello Norbert,
Thanks for your answer.
I've changed the wire:
![Screenshot 2022-03-07 at 14-05-07](https://user-images.githubusercontent.com/47952051/157040507-9cf74fe8-8306-414e-a86f-55105b74495b.jpg)
But the ESP doesn't work like it shall.
It is eqaul to wire C" to D3 or D4 and configure it right or wrong. The shown temperature is still 988.8.
I think there must be another misstake in my ideas. Can it be, that I've some wrong parameters in my configuration or that I've forgotten one?
Regards from Troisdorf
Werner

**Comment by curzon01** [comment, marked as answer]
yes, MISO and MOSI are also swapped - your attached code (which you claim it works) uses
```
D3 GPIO0  -> MX31865 CS
D6 GPIO12 -> SSPI MOSI
D7 GPIO13 -> SSPI MISO
D5 GPIO14 -> SSPI CLK
```
define that appropriately identical, then it will most likely work

**Reply by werner53844** [reply]
Thank you for your support. It works.
